length/functions/mean_squared_error.py

```diff
@@ -12,40 +13,36 @@ class MeanSquaredError(Function):
 
     def __init__(self):
         super().__init__()
-        # TODO: add more initialization if necessary
+        self.difference = None
 
     @staticmethod
     def create_one_hot(data, shape):
         assert len(shape) == 2, "Providing integers as second input to MSE only works with two dimensional input vectors"
-        # TODO: create a one-hot representation out of the given label vector
-        # Example: assume input is the following: [2, 3], and shape is (2, 4)
-        # the resulting vector should look like this:
-        # result = [[0, 0, 1, 0], [0, 0, 0, 1]]
-        return None
+        data_container = np.zeros(shape, dtype=constants.DTYPE)
+        data_container[np.arange(len(data)), data] = 1
+        return data_container
 
     def internal_forward(self, inputs):
         x1, x2 = inputs
 
         if np.issubdtype(x2.dtype, np.integer):
             x2 = self.create_one_hot(x2, x1.shape)
 
-        # TODO: calculate the mean squared error of x1 and x2
-        error = 0.0
-
-        return error
+        self.difference = x1 - x2
+        squared_sum = np.sum(np.square(self.difference))
+        return (squared_sum / self.difference.size).astype(constants.DTYPE),
```

Comment:
Alternative: `np.mean(np.square(self.difference))` and then the division is not required in 33